Development

Mikkelet opened the pull request.

Jonaskj committed:
blackscreen and circle start session

Jonaskj committed:
Blackscreen on timer

Jonaskj committed:
Timer - circle

Commit:
added open app functionality

GaagG committed:
blacklist button added

GaagG committed:
List of current apps - due to context error

GaagG committed:
Blacklist working - click-on in the list does not work.

GaagG committed:
Blacklist - made the apps on the list smaller

Jonaskj committed:
Added CircularSlider.java class

Commit:
merge with jonas

Commit:
merge with gabriel

Jonaskj committed:
BubbleSeekBar

mircealungu commented on app/src/main/java/com/mikkelthygesen/android/tid_bma_java/AppsDB.java:
```diff
@@ -0,0 +1,75 @@
+public class AppsDB extends Observable {
+    private static AppsDB sAppsDB;
+
+    //private Context mContext;
```
it is good practice to not have commented out code committed

Jonaskj committed:
BubbleSeekBar

Added a new SeekBar. This time more simple with values ranging from 1-20.

mircealungu commented on app/src/main/java/com/mikkelthygesen/android/tid_bma_java/CircularSlider.java:
```diff
@@ -0,0 +1,306 @@
+    }
+
+    // common initializer method
+    private void init(Context context, AttributeSet attrs, int defStyleAttr) {
```
this could be refactored. comments usually suggest method names

mircealungu commented on app/src/main/java/com/mikkelthygesen/android/tid_bma_java/CircularSlider.java:
```diff
@@ -0,0 +1,306 @@
+        mCircleCenterX = largestCenteredSquareRight / 2 + (w - largestCenteredSquareRight) / 2;
+        mCircleCenterY = largestCenteredSquareBottom / 2 + (h - largestCenteredSquareBottom) / 2;
+        mCircleRadius = smallerDim / 2 - mBorderThickness / 2 - mPadding;
+
```
it would be nice to have a rectangle data structure in a different class. then you can extract subfunctions for the square surrounding the circle

mircealungu commented on app/src/main/java/com/mikkelthygesen/android/tid_bma_java/CircularSlider.java:
```diff
@@ -0,0 +1,306 @@
+                int x = (int) ev.getX();
+                int y = (int) ev.getY();
+                if (x < mThumbX + mThumbSize && x > mThumbX - mThumbSize && y < mThumbY + mThumbSize && y > mThumbY - mThumbSize) {
+                    getParent().requestDisallowInterceptTouchEvent(true);
```
extract conditional to a method with an intention revealing name

mircealungu commented on app/src/main/java/com/mikkelthygesen/android/tid_bma_java/ListOfAppsOnDevice.java:
```diff
@@ -0,0 +1,157 @@
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_blacklist_recyclerview,container,false);
+
+        //Skabelsen af det interactive recyclerview.
```
keep comments in english for an open source project like this

mircealungu reviewed (approved):
PR approved
remember to address all the comments in this review
please in the future add a readme file

Jonaskj committed:
Update fragment_timer.xml

Jonaskj committed:
Background Black on app

Jonaskj committed:
Background white

Jonaskj committed:
Merge branch 'jonas' into Development

Jonaskj committed:
Update BlackList.java

Jonaskj committed:
Update fragment_black_list.xml

Jonaskj committed:
Renaming Seekbar

Jonaskj committed:
Database

GaagG committed:
changes to structure

GaagG committed:
Made a new class - BlackListAdapter - Not done yet

Still working on making the code fully functional for the BlackList.

GaagG committed:
working code now..

GaagG committed:
Now shows both lists

Only needs to merge with mamsens DB and make the check thing work.

GaagG committed:
Merge branch 'Gabriel' of https://github.com/Mikkelet/tid-bma-java into Gabriel

GaagG committed:
added a remove button in the blacklist

Had a bug where it refreshed the view while the user was interacting with it.

GaagG committed:
forgot to add a method

GaagG committed:
Checkboxes are now functional

Blacklist now only needs to be connected to the database!

GaagG committed:
cleaning up in imports

GaagG committed:
Checkboxes acts as they are supposed

Only thing left is to get the database connected. Room however is not working with me.

GaagG committed:
Delete AppName.java

Commit:
implemented shared prferences

Jonaskj committed:
Both seekbars are working

Jonaskj committed:
refactoring names and commenting

Commit:
merge with jonas

Commit:
Merge remote-tracking branch 'origin/jonas' into HEAD

Commit:
merge
